airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```diff
@@ -115,9 +116,18 @@ def _wrap_refresh_token_exception(self, exception: requests.exceptions.RequestEx
     def _get_refresh_access_token_response(self) -> Any:
         try:
             response = requests.request(method="POST", url=self.get_token_refresh_endpoint(), data=self.build_refresh_request_body())
-            self._log_response(response)
-            response.raise_for_status()
-            return response.json()
+            if response.ok:
+                response_json = response.json()
+                # Add the access token to the list of secrets so it is replaced before logging the response
+                # An argument could be made to remove the prevous access key from the list of secrets, but unmasking values seems like a security incident waiting to happen...
+                access_key = response_json[self.get_access_token_name()]
```

Comment:
nit: not sure if this is a legitimate concern, but if for whatever reason the API has a bug and the access token key doesn't exist in the response this will raise a not so helpful `KeyError`. Maybe while we're fixing this we add a nicer error message like:
```
access_key = response_json.get(self.get_access_token_name())
if not access_key:
  raise Exception("Token refresh API response was missing access token {self.get_access_token_name()}")